frontend/app/src/App.tsx

```diff
@@ -1153,13 +1153,7 @@ export class App extends PureComponent<Props, State> {
       return "hash_for_undefined_custom_theme"
     }
 
-    const themeInputEntries = Object.entries(themeInput)
-    // Ensure that our themeInput fields are in a consistent order when
-    // stringified below. Sorting an array of arrays in javascript sorts by the
-    // 0th element of the inner arrays, uses the 1st element to tiebreak, and
-    // so on.
-    themeInputEntries.sort()
-    return hashString(themeInputEntries.join(":"))
+    return hashString(JSON.stringify(themeInput))
```

Comment:
The method before only worked on a flat structure. This should also work on a nested structure to detect changes in the theme.